extensions/sql-migration/src/models/loginMigrationModel.ts

```diff
@@ -0,0 +1,247 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as mssql from 'mssql';
+import { MultiStepResult, MultiStepState } from '../dialog/generic/multiStepStatusDialog';
+import * as constants from '../constants/strings';
+import { LoginTableInfo } from '../api/sqlUtils';
+
+type ExceptionMap = { [login: string]: any }
+
+export enum LoginMigrationStep {
+	NotStarted = -1,
+	MigrateLogins = 0,
+	EstablishUserMapping = 1,
+	MigrateServerRolesAndSetPermissions = 2,
+	MigrationCompleted = 3,
+}
+
+export function GetLoginMigrationStepString(step: LoginMigrationStep): string {
+	switch (step) {
+		case LoginMigrationStep.NotStarted:
+			return constants.NOT_STARTED;
+		case LoginMigrationStep.MigrateLogins:
+			return constants.MIGRATE_LOGINS;
+		case LoginMigrationStep.EstablishUserMapping:
+			return constants.ESTABLISH_USER_MAPPINGS;
+		case LoginMigrationStep.MigrateServerRolesAndSetPermissions:
+			return constants.MIGRATE_SERVER_ROLES_AND_SET_PERMISSIONS;
+		case LoginMigrationStep.MigrationCompleted:
+			return constants.LOGIN_MIGRATION_COMPLETED;
+		default:
+			return ""
+	}
+}
+
+export enum LoginMigrationState {
```

Comment:
adding to my other comment, even if there isn't a preexisting enum that captures all these states, at least we could definitely consolidate the two between LoginMigrationState and MultiStepState since they seem identical

Reply:
yeah agreed. i was struggling over what to do here. on one hand, i wanted the dialog to be generic enough so it can be reused easily. on the other hand, I wasn't sure if it would be good design to couple the login migration states to it as opposed to have an explicit conversion. Seems like you're leaning towards have just 1 enum?

Reply:
consolidated into 1 enum